pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -682,6 +684,43 @@ func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispa
 
 	memory := resource.MustParse(m.Spec.Container.Memory)
 	replicas := m.Spec.Replicas
+	volumeMounts := []corev1.VolumeMount{{
+		Name:      ConfigVolumeName,
+		MountPath: consts.ServerConfigRoot,
+	}, {
+		Name:      dataVolumeName,
+		MountPath: DataMountPath,
+	}}
+	volumes := []corev1.Volume{{
+		Name: ConfigVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: configMap.Name},
+			},
+		},
+	}}
+
+	if m.HasCustomLibraries() {
+		// Persistent volume claim for custom libraries
+		customPvc := &corev1.PersistentVolumeClaim{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: m.Namespace, Name: fmt.Sprintf("%s-custom-libraries", m.Name)}, customPvc)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info(fmt.Sprintf("Creating '%s'", m.Spec.Dependencies.CustomLibrariesPVName))
+				customPvc.Name = fmt.Sprintf("%s-custom-libraries", m.Name)
+				customPvc.Namespace = m.Namespace
+				customPvc.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{"ReadOnlyMany"}
+				customPvc.Spec.Resources.Requests = corev1.ResourceList{corev1.ResourceStorage: consts.DefaultPVSize}
+				customPvc.Spec.VolumeName = m.Spec.Dependencies.CustomLibrariesPVName
```

Comment:
Instead of updating individual fields on `customPvc` can you update this to create a new object like we do elsewhere as it makes it much easier to read. e.g:

```go
customPvc = corev1.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{
			Name:      ...,
			Namespace: m.Namespace,
		},
			Spec: corev1.PersistentVolumeClaimSpec{
				AccessModes: []corev1.PersistentVolumeAccessMode{
					corev1.ReadOnlyMany,
				},
				Resources: corev1.ResourceRequirements{
					Requests: corev1.ResourceList{
						corev1.ResourceStorage: consts.DefaultPVSize,
					},
				},
			},
		}
```

Reply:
I need to pass that instance to the client.Get method, is it safe to do? I mean on not found my instance is returned unchanged?

Reply:
Seems to a be a problem if we need to update it, not a problem with `Get` and `Create` however `CreateOrUpdate` is a an option or two entries with `ObjectMeta` only and another whole structure to Create

Reply:
changed creation to literals. hard to say if go readability has been improved :D